common/lib/dependabot/config/ignore_condition.rb

```diff
@@ -50,10 +50,10 @@ def ignore_patch(version)
         parts = version.split(".")
         return [] unless parts.size > 2
 
-        lower_parts = parts.first(2) + ["a"]
+        lower_parts = parts.first(3)
```

Comment:
Question: can we drop this `a`?

For context, it's a metacharacter recognized by [Gem::{Version.Requirement}](https://docs.ruby-lang.org/en/3.0.0/Gem/Version.html), which we is the implementation of semver used for most ecosystems.

> If any part contains letters (currently only a-z are supported) then that version is considered prerelease. Versions with a prerelease part in the Nth part sort less than versions with N-1 parts. Prerelease parts are sorted alphabetically using the normal Ruby string sorting rules. If a prerelease part contains both letters and numbers, it will be broken into multiple parts to provide expected sort behavior (1.0.a10 becomes 1.0.a.10, and is greater than 1.0.a9).

I ask because I just cargo culted it along in the base implementation, the change makes sense but I didn't dig into the implications.

Reply:
Good question! I haven't yet thought of a case where it would matter but @feelepxyz figured out what it would look like to keep the `a` here:
https://github.com/dependabot/dependabot-core/blob/2dddae6b97cbef2846d31e4ceecc9f4fbacf0714/common/spec/dependabot/config/ignore_condition_spec.rb#L89